config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,20 +924,24 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
+        // this has two possible outcomes depending on whether
+        // we resolve and memoize a first or b first. currently
+        // java 8's hash table makes it resolve OK, but
```

Comment:
this seems kind of like a cop out :-)  i'm not sure we should have an undefined behavior for this.  i would think it shouldn't be allowed to both resolve okay and throw an exception

i think actually if you change the HashMap mentioned in https://github.com/typesafehub/config/issues/253 to a LinkedHashMap then you'll always get this behavior across Java versions including both Java 7 and Java 8. (note that'll make one other test fail, but as far as I can tell, if you're going to change this test then that one should be changed too). i do think it'd be good to have just one behavior allowed here whatever that behavior is. so maybe change the HashMap to a LinkedHashMap and remove this comment?